src/rpc/protocol/payloads.rs

```diff
@@ -638,15 +716,44 @@ impl ObTableBatchOperation {
             row_keys,
             Some(columns),
             Some(properties),
+            None,
+            None,
         ));
     }
 
+    /// check the data with corresponding row_keys
+    /// whether meet the filter and execute the insertUp
+    /// if check_exist is true: check if any data meet the filter
+    /// if check_exist is false: check if all data do not meet the filter
+    pub fn check_and_insert_up<T: FilterEncoder>(
+        &mut self,
+        row_keys: Vec<Value>,
+        columns: Vec<String>,
+        properties: Vec<Value>,
+        filter: T,
+        check_exist: bool,
+    ) {
```

Comment:
How about this name:
```suggestion
    pub fn check_and_upsert(
        &mut self,
        row_keys: Vec<Value>,
        columns: Vec<String>,
        properties: Vec<Value>,
        filter: impl FilterEncoder,
        check_exist: bool,
    ) {
```